### ISSUE TITLE
buildx does not respect architecture selection causing slow builds

### ISSUE THREAD

**Issue by Firstyear** [docker/buildx#1686]
### Contributing guidelines

- [X] I've read the [contributing guidelines](https://github.com/docker/buildx/blob/master/.github/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

When using docker buildx with a cluster of machines, where one machine is arm64 and another is x86_64, docker buildx will always send builds to the incorrect arch worker. For example, arm64 builds are sent to the x86_64 machine. 

```
NAME/NODE       DRIVER/ENDPOINT                         STATUS  BUILDKIT PLATFORMS
cluster *       docker-container
  cluster2      tcp://HOSTNAME:2376 running v0.10.5  linux/amd64*, linux/amd64/v2*, linux/amd64/v3*, linux/amd64/v4*, linux/arm64, linux/riscv64, linux/ppc64le, linux/s390x, linux/386, linux/mips64le, linux/mips64, linux/arm/v7, linux/arm/v6
  cluster3      unix:///var/run/docker.sock             running v0.11.2  linux/arm64*, linux/amd64, linux/amd64/v2, linux/riscv64, linux/ppc64le, linux/s390x, linux/386, linux/mips64le, linux/mips64, linux/arm/v7, linux/arm/v6
```

Builds for arm64 are always sent to cluster2 instead of cluster3 in this example. 

### Expected behaviour

Builds should be sent to the works that are preferred for those arches (and marked with a *). 

### Actual behaviour

Builds are sent to the incorrect host that then runs under qemu (and very slowly). 

### Buildx version

github.com/docker/buildx v0.10.0 876462897612d36679153c3414f7689626251501

### Docker info

```markdown
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc., v0.10.0)
  compose: Docker Compose (Docker Inc., v2.15.1)
  dev: Docker Dev Environments (Docker Inc., v0.0.5)
  extension: Manages Docker extensions (Docker Inc., v0.2.17)
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc., 0.6.0)
  scan: Docker Scan (Docker Inc., v0.23.0)

Server:
 Containers: 7
  Running: 1
  Paused: 0
  Stopped: 6
 Images: 10
 Server Version: 20.10.22
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runtime.v1.linux runc io.containerd.runc.v2
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 9ba4b250366a5ddde94bb7c9d1def331423aa323
 runc version: v1.1.4-0-g5fd4c4d
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
  cgroupns
 Kernel Version: 5.15.49-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: aarch64
 CPUs: 8
 Total Memory: 7.765GiB
 Name: docker-desktop
 ID: FJ2Q:E2IS:QP4N:ZSNS:VIEN:2MBL:EHCI:RGXI:DJJG:5DUD:YVH4:QUI6
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: true
 Insecure Registries:
  hubproxy.docker.internal:5000
  127.0.0.0/8
 Live Restore Enabled: false
```
```


### Builders list

```markdown
NAME/NODE       DRIVER/ENDPOINT                         STATUS  BUILDKIT PLATFORMS
cluster *       docker-container
  cluster2      tcp://HOSTNAME:2376 running v0.10.5  linux/amd64*, linux/amd64/v2*, linux/amd64/v3*, linux/amd64/v4*, linux/arm64, linux/riscv64, linux/ppc64le, linux/s390x, linux/386, linux/mips64le, linux/mips64, linux/arm/v7, linux/arm/v6
  cluster3      unix:///var/run/docker.sock             running v0.11.2  linux/arm64*, linux/amd64, linux/amd64/v2, linux/riscv64, linux/ppc64le, linux/s390x, linux/386, linux/mips64le, linux/mips64, linux/arm/v7, linux/arm/v6
```
```


### Configuration

```markdown
n/a
```

**Comment by tonistiigi** [comment]
Is there an output that makes you think that the process is on a wrong node? The `buildx inspect` output looks correct to me.

**Comment by Firstyear** [comment]
The output is annoyingly "I log into my x86_64 server and check it's cpu and it's running qemu-aarch64 and the build takes about 10 hours, while my aarch64 system has 0% cpu usage and is idle". 

The docker buildx ls absolutely looks correct, but it really isn't selecting the correct nodes.

**Comment by Firstyear** [comment]
I wonder if the following is related? https://github.com/docker/buildx/issues/1519

When I force the builders to have qemu tools removed to prevent the incorrect selection, I now get the error for manifest merging as documented here. Could this be involved?

i'm happy to repro and send other details btw :)

**Comment by tonistiigi** [comment]
I don't think so. But it might be some `Node` datastructure refactoring PRs @crazy-max

**Comment by Firstyear** [comment]
Well, currently I think there is no output in the --progress to tell you where a job/task is running, would it be a simple-ish change to add an "internal" info line, similar to how at the start of builds you have:

```
#3 [linux/arm64 internal] load metadata for docker.io/opensuse/tumbleweed:latest
```

Could there be one that says `#3 [linux/arm64 internal] using builder <name>`

**Comment by crazy-max** [comment]
> github.com/docker/buildx v0.10.0 [8764628](https://github.com/docker/buildx/commit/876462897612d36679153c3414f7689626251501)

It has been fixed in [0.10.2](https://github.com/docker/buildx/releases/tag/v0.10.2)

**Comment by Firstyear** [comment]
Thank you @crazy-max !
